confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.confignode.client;
+
+import org.apache.iotdb.common.rpc.thrift.TEndPoint;
+import org.apache.iotdb.common.rpc.thrift.TSStatus;
+import org.apache.iotdb.commons.client.IClientManager;
+import org.apache.iotdb.commons.client.sync.SyncConfigNodeIServiceClient;
+import org.apache.iotdb.confignode.rpc.thrift.TConfigNodeLocation;
+import org.apache.iotdb.confignode.rpc.thrift.TConfigNodeRegisterReq;
+import org.apache.iotdb.confignode.rpc.thrift.TConfigNodeRegisterResp;
+import org.apache.iotdb.db.client.DataNodeClientPoolFactory;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Sync send RPC requests to ConfigNode. See confignode.thrift for more details. */
+public class SyncConfigNodeClientPool {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SyncConfigNodeClientPool.class);
+
+  private static final int retryNum = 5;
```

Comment:
I recommend adding some backoff sleep strategies for retry, such as a slight sleep of 100ms, otherwise the retry is likely to fail again. BTW, you can abstract the retry logic into one place, see [example](https://carlosbecker.com/posts/exponential-backoff-java8/). Of course, this work can be optimized later, so please add at least one TODO

Reply:
Fixed and added TODO.